Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/WatchManager.java

```diff
@@ -474,6 +650,84 @@ private void updateState(boolean watchEnabled, boolean backgroundEnabled) {
         }
     }
 
+    private void updateDeletedOrArchivedPins() {
+        for (Pin pin : pins) {
+            if (pin.isError || pin.archived || !pin.watching) {
+                SavedThread savedThread = findSavedThreadByLoadableId(pin.loadable.id);
+                if (savedThread == null || savedThread.isStopped || savedThread.isFullyDownloaded) {
+                    continue;
+                }
+
+                if (pin.isError || pin.archived) {
+                    savedThread.isFullyDownloaded = true;
+                }
+
+                if (!savedThread.isFullyDownloaded) {
+                    continue;
+                }
+
+                // TODO: should the pinType be updated here?
+                createOrUpdateSavedThread(savedThread);
+
+                databaseManager.runTask(() -> {
+                    // Update thread in the DB
+                    if (savedThread.isStopped) {
+                        databaseManager.getDatabaseSavedThreadmanager()
+                                .updateThreadStoppedFlagByLoadableId(pin.loadable.id, true).call();
+                    }
+
+                    // Update thread in the DB
+                    if (savedThread.isFullyDownloaded) {
+                        databaseManager.getDatabaseSavedThreadmanager()
+                                .updateThreadFullyDownloadedByLoadableId(pin.loadable.id).call();
+                    }
+
+                    databasePinManager.updatePin(pin).call();
```

Comment:
Do the same thing when unpinning or stopping to download a thread (the two toolbar buttons in a thread)

Reply:
Apparently not needed anymore.